### ISSUE TITLE
Access Violation in Microsoft.Data.SqlClient.SNI build 5.0.0

### ISSUE THREAD

**Issue by rozeboosje** [dotnet/SqlClient#1693]
### Describe the bug
Our application - .NET Framework 4.8 - uses Microsoft.Data.SqlClient and this also pulled in Microsoft.Data.SqlClient.SNI build 4.0.0

Checking the NuGet Package Manager for our solution I see there is an upgrade to build 5.0.0 of Microsoft.Data.SqlClient.SNI so I installed that. Upon which the software throws an AccessViolation when opening a connection.

If you are seeing an exception, include the full exceptions details (message and stack trace).

```
Exception message: Exception thrown: 'System.AccessViolationException' in Microsoft.Data.SqlClient.dll
Stack trace:
```

### To reproduce
As described above

```

### Expected behavior
No Access Violation? 

### Further technical details
Microsoft.Data.SqlClient version: 4.1.0 
.NET target: Framework 4.8
SQL Server version: Azure SQL

**Additional context**
I suspect that Microsoft.Data.SqlClient.SNI build 5.0.0 might need to be included with an update to a build 5.something for Microsoft.Data.SqlClient which is not yet available, and that if you run build 5.0.0 against build 4.1.0 of Microsoft.Data.SqlClient the two will not be mutually compatible....

.... or something like that?

**Comment by ErikEJ** [comment]
You cannot do that and you should never reference the SNI package explicitly

**Comment by JRahnama** [comment]
@rozeboosje as @ErikEJ mentioned you should never reference the SNI package explicitly. Each version of MDS has a dependency to to a specific build of SNI, usually they have the same version number. The SNI v5.0 is released to make it accessible for Microsoft.Data.SqlClient v5.0 which is going to be released today or tomorrow.

**Comment by rozeboosje** [comment]
Yeah that's fair enough and a similar message is shown in the NuGet manager ... but I don't even recall explicitly getting that package. I reckon it automatically got included when I installed the MDS package, but it's presented as a separate package and prompted me to upgrade so I did... would it not be better if MDS and the SNI were released together as part of a single package then? But no worries. I rolled it back and it all came good again.
